Create SeaFood UseCase test case

AbdElrahman-Rafaat-Amer opened the pull request:
- Created SeaFood UseCase test case

AbdElrahman-Rafaat-Amer committed:
- write test cases for SeaFood UseCase

Took 2 hours 27 minutes

AbdElrahman-Rafaat-Amer committed:
Merge remote-tracking branch 'origin/development' into test/seafood-test

# Conflicts:
#	build.gradle

Took 2 minutes

AbdElrahman-Rafaat-Amer committed:
- removed duplicate file MealsDummy.kt

Took 3 minutes

ahmednasserzaza commented on src/main/kotlin/domain/usecase/GetSeafoodMealsUseCase.kt:
```diff
@@ -10,7 +10,7 @@ class GetSeafoodMealsUseCase(private val mealRepository: MealRepository) {
         }.sortedByDescending {
             it.nutrition.protein!!
```
Please avoid using not null assertion

AbdElrahman-Rafaat-Amer replied:
it's already filtered to not have null name or protein
```
mealRepository.getMeals().filter {
            it.tags.contains(SEAFOOD) && it.mealName != null && it.nutrition.protein != null
        }
```

ahmednasserzaza commented on src/main/kotlin/domain/usecase/GetSeafoodMealsUseCase.kt:
```diff
@@ -10,7 +10,7 @@ class GetSeafoodMealsUseCase(private val mealRepository: MealRepository) {
             it.nutrition.protein!!
         }.map { seafoodMeal ->
-            seafoodMeal.mealName!! to (seafoodMeal.nutrition.protein ?: 0f)
+            seafoodMeal.mealName!! to seafoodMeal.nutrition.protein!!
```
same here

AbdElrahman-Rafaat-Amer replied:
it's already filtered to not have null name or protein
```
mealRepository.getMeals().filter {
            it.tags.contains(SEAFOOD) && it.mealName != null && it.nutrition.protein != null
        }
```

ahmednasserzaza commented on src/test/kotlin/domain/usecase/GetSeafoodMealsUseCaseTest.kt:
```diff
@@ -0,0 +1,142 @@
+    @Test
+    fun `should return Failure Result of NoMealsFoundException when can't find any meal that contain seafood in tags`() {
+        //Given
+        every { mealsRepository.getMeals() } returns listOf(
```
I think it should be better if you made those sample of data into other util class and call it here or simply add it into the same test class but in companion object

ahmednasserzaza commented on src/test/kotlin/domain/usecase/GetSeafoodMealsUseCaseTest.kt:
```diff
@@ -0,0 +1,142 @@
+        //Given
+        every { mealsRepository.getMeals() } returns listOf(
+            createMeal(
+                tags = listOf("seafood"),
```
Isn't that a seafood !

AbdElrahman-Rafaat-Amer replied:
But name is null + protein also null

AbdElrahman-Rafaat-Amer replied:
Updated to be more readable and match the test

ahmednasserzaza commented on src/test/kotlin/domain/usecase/GetSeafoodMealsUseCaseTest.kt:
```diff
@@ -0,0 +1,142 @@
+    fun `should return Failure Result of NoMealsFoundException when all seafoods meals name ar null`() {
+        //Given
+        every { mealsRepository.getMeals() } returns listOf(
+            createMeal(
```
Where is the null value for name here ?
I think its the default value, but anyway you should specify it here in test case to be more accurate and not confusing

ahmednasserzaza commented on src/test/kotlin/domain/usecase/GetSeafoodMealsUseCaseTest.kt:
```diff
@@ -0,0 +1,142 @@
+    }
+
+    @Test
+    fun `should return Success Result List of Pair MealName & protein when all meals contain seafoods in tags and Name & Protein ar not null`() {
```
I think the test case naming could be improved and also there is a typo here

ahmednasserzaza reviewed (approved):
Well done 
Just check the comments please

AbdElrahman-Rafaat-Amer committed:
Merge remote-tracking branch 'origin/development' into test/seafood-test

AbdElrahman-Rafaat-Amer committed:
- applied comments enhancements.
- made the function name more readable.
- extracted the meals to MockMeals.kt

Took 24 minutes

AbdElrahman-Rafaat-Amer committed:
Merge remote-tracking branch 'origin/development' into test/seafood-test
